src/components/CardSchemas/ProductCard.jsx

```diff
@@ -1,22 +1,49 @@
 import { Link } from "react-router-dom"
+import { useContext, useEffect } from "react";
+import { AppContext } from "../../App";
 import styled from "styled-components"
 import noImage from "../../assets/images/defaultProductImage.png"
 
-export default function ProductCard({title}){
+export default function ProductCard({product, admin}){
+
+    const {dispatch} = useContext(AppContext)
+
+    useEffect(()=>{
+
+    },[product.disabled])
+
+    function thousandsFormat(num){
+        return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ".")
+    }
     return (
-        <Card>
+        <Card className='wrapper'>
             <Wrapper>
-                <Image src={noImage} alt="noImage"/>
+                {admin
+                ?<Options>
+                    <Toggle toggled={product.disabled} onClick={()=>dispatch({type:'toggleProduct', id:product.id})}>{product.disabled?'Disabled':'Enabled'}</Toggle>
+                    <Link to={`edit/${product.id}`}>
+                        <Toggle>Edit</Toggle>
+                    </Link>
+                </Options>
+                :null}
+                <Image src={product?.image||noImage} alt="noImage" onError={e=>e.target.src=noImage}/>
                 <hr/>
-                <h1>{title}</h1>
+                <Data>
+                    <Price>{(product.price&&'$'+thousandsFormat(product.price))||'Price'}</Price>
```

Comment:
Yes, js won't allow for symbols other than '_' in variable naming, so it distinguishes between variables and operands in the same way it would distinguish them in the operation '$'+thousandsFormat(), although I should be add spaces for legibility

Reply:
Yes, this makes the code difficult to read. In a team where you code is constantly reviewed and also updated by other developers. You need to make sure it's readable.